Zend/Zend.m4

```diff
@@ -1,5 +1,15 @@
 dnl This file contains Zend specific autoconf macros.
 
+AC_DEFUN([ZEND_CHECK_TYPEDEF_REDECL], [dnl
+AC_CACHE_CHECK([if compiler allows typedef redeclarations],
+  [php_cv_have_typedef_redecl],
+  [AC_LINK_IFELSE([AC_LANG_PROGRAM([], [dnl
+    typedef int mytype;
+    typedef int mytype;
```

Comment:
```suggestion
    typedef int mytype;
    mytype var;
    (void)var;
```

Just in case some configuration of clang will have issues with unused local typdefs (-Wunused-local-typedef) and/or unused variables (-Wunused-variable).